speculation-rules.bs

```diff
@@ -627,6 +629,97 @@ A <dfn>prerender candidate</dfn> is a [=struct=] with the following [=struct/ite
+      The <a>`type`</a> string is set to
+      "<dfn dfn-type="grammar" link-type="grammer">`script speculationrules`</dfn>"
+      if the {{Element}} is {{HTMLScriptElement}} and the {{HTMLScriptElement/type}}
+      attribute is "`speculationrules`", or set to
+      "<dfn dfn-type="grammar" link-type="grammer">`script attribute`</dfn>" for
+      JavaScript event handlers. Otherwise, it is set to 
+      "<dfn dfn-type="grammar" link-type="grammer">`script`</dfn>".
+
+<h3 id="content-security-policy-patches-source-lists">Source Lists</h3>
+In the [[CSP#framework-directive-source-list]], we change the following definition to have a new source, `inline-speculation-rules`.
+
+<pre dfn-type="grammar" link-type="grammar">
+  <dfn>keyword-source</dfn> = "<dfn>'self'</dfn>" / "<dfn>'unsafe-inline'</dfn>" / "<dfn>'unsafe-eval'</dfn>"
+    / "<dfn>'strict-dynamic'</dfn>" / "<dfn>'unsafe-hashes'</dfn>"
+    / "<dfn>'report-sample'</dfn>" / "<dfn>'unsafe-allow-redirects'</dfn>"
+    / "<dfn>'wasm-unsafe-eval'</dfn>" / "<dfn>'inline-speculation-rules'</dfn>"
+</pre>
+
+<h3 id="content-security-policy-patches-should-block-inline">Should element's inline type behavior be blocked by Content Security Policy?</h3>
+Also, we make a following change for a note in [[CSP#should-block-inline]].
+
+Note: The valid values for <a>`type`</a> are "`script`", "`script attribute`", "`script speculationrules`", "`style`", and "`style attribute`".
+
+<h3 id="content-security-policy-patches-script-src">script-src</h3>
+Also we make the following change for [[CSP#directive-script-src]].
+
+  3.  Inline <{script}> blocks MUST pass through [[CSP#should-block-inline]]. Their behavior will be blocked unless every policy allows inline script, either implicitly by not specifying a `script-src` (or `default-src`) directive, or explicitly, by specifying "`unsafe-inline`", a <a grammar>nonce-source</a> or a <a grammar>hash-source</a> that matches the inline block, or specifying "`inline-speculation-rules`" on the type attribute being "`speculationrules`".
+
+<h3 id="content-security-policy-patches-allow-all-inline">Does a source list allow all inline behavior for type?</h3>
+The algorighm needs a patch to handle the "`script speculationrules`" type.
+
+  2.  <a for=set>For each</a> |expression| of list:
+
+      1.  If |expression| matches the <a grammar>`nonce-source`</a> or
+          <a grammar>`hash-source`</a> grammar, return "`Does Not Allow`".
+
+      2.  If <a>`type`</a> is "`script`", "`script attribute`" or "`navigation`"
+          and |expression| matches the <a grammar>keyword-source</a>
+          "<a grammar>`'strict-dynamic'`</a>", return "`Does Not Allow`".
+
+          Note: `'strict-dynamic'` only applies to scripts, not other resource
+          types. Usage is explained in more detail in [[CSP#strict-dynamic-usage]].
+
+      3.  If <a>`type`</a> is "`script speculationrules`" and |expression| matches the
```

Comment:
For inserted things inside multi-step algorithms, it would be good to mark them up with `<ins>` / `</ins>` to make it clearer to the reader.

Reply:
Resolved

Reply:
Love this btw. I'd totally forgetten `<ins>` even existed.